neurodamus/node.py

```diff
@@ -377,37 +376,27 @@ def compute_load_balance(self):
             logging.info("Load Balancing ENABLED. Mode: Memory")
             filename = f"allocation_r{MPI.size}_c{SimConfig.modelbuilding_steps}.pkl.gz"
 
-            if ospath.exists(filename):
-                alloc = import_allocation_stats(filename, self._cycle_i)
+            file_exists = ospath.exists(filename)
+            MPI.barrier()
```

Comment:
The barrier there was put to avoid a race condition. In that part of the code we are checking if the allocation file is there or not. If the file is not there it gets created just a few lines later. We could be in a condition where some of the ranks are still checking for the existence of the file while others are already creating it, so we could have some ranks passing the check because some of the other ranks on the same run have already created the file.

The race condition never happened, actually, but I've introduced it as a safety measure. If you think of any good reason why it shouldn't be there I can remove it, otherwise it's just a way to be sure the race condition never happens at no performance cost.